src/ledger/LedgerManagerImpl.cpp

```diff
@@ -947,6 +951,21 @@ LedgerManagerImpl::closeLedger(LedgerCloseData const& ledgerData)
     // step 4
     mApp.getBucketManager().forgetUnreferencedBuckets();
 
+    // Spin for parametrized amount of time in simulation mode
+    if (mApp.getMode() == Application::AppMode::RUN_CONSENSUS_ONLY)
```

Comment:
I don't think you should be double guarding this setting with "mode":
I would simply have a setting `DELAY_TX_APPLY_MS_FOR_TESTING` that if non 0 causes an additional sleep